controllers/sync.go

```diff
@@ -529,7 +529,11 @@ func (r *YtsaurusReconciler) Sync(ctx context.Context, resource *ytv1.Ytsaurus)
 		if !componentManager.needSync() {
 			logger.Info("Ytsaurus has synced and is running now")
 			err := ytsaurus.SaveClusterState(ctx, ytv1.ClusterStateRunning)
-			return ctrl.Result{Requeue: true}, err
+			if err != nil {
+				return ctrl.Result{}, err
+			}
+			err = ytsaurus.SaveObserverGeneration(ctx, resource.ObjectMeta.Generation)
+			return ctrl.Result{}, err
```

Comment:
AFAIK it is a bad practice to request requeue by by the operator, though let's leave `Requeu: true` here as was before, since it is unintended/unrelated to the current issue change in behaviour.